pykube/http.py

```diff
@@ -32,13 +32,15 @@ def build_session(self):
         s = requests.Session()
         if "certificate-authority" in self.config.cluster:
             s.verify = self.config.cluster["certificate-authority"].filename()
+
         if "token" in self.config.user and self.config.user["token"]:
             s.headers["Authorization"] = "Bearer {}".format(self.config.user["token"])
-        else:
+        elif "client-certificate" in self.config.user:
```

Comment:
I am also considering logging a warning (which can be ignored and that's fine) in the `else` branch indicating that we are not adding any sort of authentication information. This would be helpful for those who do intend to add it or stay as a reminder that it might be a good idea to enable it. 

I'd hate for someone to think that just because they can access the API server means it's ready for production.